src/Sentry/Platforms/Cocoa/SentrySdk.cs

```diff
@@ -88,27 +89,44 @@ private static void InitSentryCocoaSdk(SentryOptions options)
 
         // TODO: Finish SentryEventExtensions to enable these
 
-        // if (options.Native.EnableBeforeSend && options.BeforeSend is { } beforeSend)
-        // {
-        //     nativeOptions.BeforeSend = evt =>
-        //     {
-        //         var sentryEvent = evt.ToSentryEvent(nativeOptions);
-        //         var result = beforeSend(sentryEvent)?.ToCocoaSentryEvent(options, nativeOptions);
-        //
-        //         // Note: Nullable result is allowed but delegate is generated incorrectly
-        //         // See https://github.com/xamarin/xamarin-macios/issues/15299#issuecomment-1201863294
-        //         return result!;
-        //     };
-        // }
-
-        // if (options.Native.OnCrashedLastRun is { } onCrashedLastRun)
-        // {
-        //     nativeOptions.OnCrashedLastRun = evt =>
-        //     {
-        //         var sentryEvent = evt.ToSentryEvent(nativeOptions);
-        //         onCrashedLastRun(sentryEvent);
-        //     };
-        // }
+        if (options.Native.BeforeSend is { } beforeSend)
```

Comment:
I think this needs reconciling/merging with:
https://github.com/getsentry/sentry-dotnet/blob/17790c77a123806f732d7563543ecd3d5ae84cfc/src/Sentry/Platforms/Cocoa/SentrySdk.cs#L169-L201

Reply:
Done - I made sure this code runs before the user BeforeSend